src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,23 +1233,126 @@ private int SkipBlockComment(int i)
+                case 'r': return "\r";
+                case 't': return "\t";
+                case 'u': return ScanUnicodeEscapeSequence();
+                case 'v': return "\v";
+                default: return c.ToString();
+            }
+        }
+
+        private string ScanUnicodeEscapeSequence()
+        {
+            int escSeqStartIndex = _currentIndex - 2;
+            int maxNumberOfHexDigits = 6;
+            bool isTerminated = false;
+
+            char c = GetChar();
+            if (c != '{')
+            {
+                UngetChar();
+
+                IScriptExtent errorExtent = NewScriptExtent(escSeqStartIndex, _currentIndex);
+                ReportError(errorExtent, () => ParserStrings.InvalidUnicodeEscapeSequence);
+                return s_unknownUnicodeChar;
+            }
+
+            // Scan hex chars after the Unicode escape sequence start.
+            var sb = GetStringBuilder();
+            int i;
+            for (i = 0; i < maxNumberOfHexDigits; i++)
+            {
+                c = GetChar();
+
+                // Sequence has been terminated.
+                if (c == '}')
+                {
+                    if (i == 0)
+                    {
+                        // Sequence must have at least one hex char.
+                        IScriptExtent errorExtent = NewScriptExtent(escSeqStartIndex, _currentIndex);
+                        ReportError(errorExtent, () => ParserStrings.InvalidUnicodeEscapeSequence);
+                        return s_unknownUnicodeChar;
+                    }
+
+                    isTerminated = true;
+                    break;
+                }
+
+                if (!c.IsHexDigit())
+                {
+                    UngetChar();
+
+                    ReportError(_currentIndex, () => ParserStrings.MissingUnicodeEscapeSequenceTerminator);
+                    return s_unknownUnicodeChar;
+                }
+
+                sb.Append(c);
+            }
+
+            if (!isTerminated)
+            {
+                c = GetChar();
+                if (c != '}')
+                {
+                    UngetChar();
+                    ReportError(_currentIndex,
+                        i == maxNumberOfHexDigits && c.IsHexDigit()
+                            ? (Expression<Func<string>>)(() => ParserStrings.TooManyDigitsInUnicodeEscapeSequence)
+                            : () => ParserStrings.MissingUnicodeEscapeSequenceTerminator);
+                    return s_unknownUnicodeChar;
+                }
+            }
+
+            string hexStr = GetStringAndRelease(sb);
+
+            int unicodeValue = int.Parse(hexStr, NumberStyles.AllowHexSpecifier, NumberFormatInfo.InvariantInfo);
+            if (unicodeValue <= Char.MaxValue)
+            {
+                return ((char)unicodeValue).ToString();
+            }
+            else if (unicodeValue <= 0x10FFFF)
+            {
+                return Char.ConvertFromUtf32(unicodeValue);
```

Comment:
You misinterpret your results. What you see are runtime errors not parse time errors.  Both Bash and Python happily allow these **at parse time**.  And they **may** error at runtime.  They didn't error for me (except for the obvious parse time error I introduced to demonstrate the difference):

![image](https://user-images.githubusercontent.com/5177512/27598803-dd610c7a-5b23-11e7-83f6-72e07d19a683.png)

BTW just because the  output is gibberish in one terminal/font combo doesn't mean it will be in all.  And folks sometimes aren't generating output to the host but to a file.  Most modern editors can render Unicode chars including those specified with a surrogate char specifier like `` `u{1f44d}``.

Besides there is nothing to stop you from doing this today using sub-expressions e.g.:
![image](https://user-images.githubusercontent.com/5177512/27599137-fe609f3e-5b24-11e7-83fe-39424a8303fe.png)

Not that I suggest folks specify individual surrogate chars.  Why do that when you can just use `` `u{1F44d}``?

The basic ask here is that folks want Emoji support.  The Windows console does support Unicode and is only limited by the font support of Unicode glyphs.  And there are a number of hosts that have even better support i.e. ISE and VSCode's terminal.

Finally what we have now is basically what C# supports for Unicode escape sequence processing sans the obvious fixed width vs variable width difference.  C# allows you to specify whatever Unicode character you want regardless of whether it's a surrogate char or not.  

I'm not sure what complexity you refer to.  It certainly isn't complexity in the implementation.  To add parse time errors for certain character ranges  **would** introduce complexity.